File: spring-web-java/src/main/java/spring/web/java/presentation/converter/OrderStatusConverter.java
package spring.web.java.presentation.converter;

import org.springframework.core.convert.converter.Converter;
import org.springframework.stereotype.Component;

import spring.web.java.domain.model.enums.OrderStatus;

@Component
public class OrderStatusConverter implements Converter<String, OrderStatus> {

	@Override
	public OrderStatus convert(String source) {
		return OrderStatus.of(source);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/presentation/converter/OrderStatusConverter.kt
package spring.web.kotlin.presentation.converter

import org.springframework.core.convert.converter.Converter
import org.springframework.stereotype.Component
import spring.web.kotlin.domain.model.enums.OrderStatus

@Component
class OrderStatusConverter : Converter<String, OrderStatus> {
    override fun convert(source: String) = OrderStatus.of(source)
}